File: Guanabara - Java/Meus exercícios POO/Aula15-Exercicio_Final/src/main/java/projeto_youtube/Gafanhoto.java
package projeto_youtube;
public class Gafanhoto extends Pessoa{ //conceito de Herança: `Gafanhoto` herda `Pessoa`
    //Atributos Privados (encapsulados)
    private String login;
    private int totAssistido;
    
    //Construtor
    public Gafanhoto(String nome, int idade, char sexo, String loginn) {
        super(nome, idade, sexo); //chama o construtor da superclasse `Pessoa`
        //Atributo passado por parâmetro
        this.login = loginn;
        
        //Atributo pré-configurado
        this.totAssistido = 0;
    }
    
    //Método Público
    public void viuMaisUm() {
        
    }
    
    //Método toString sobrepondo sua superclasse // perceba a chamada do método `toString()` da superclasse `Pessoa`.
    @Override
    public String toString() {
        return "Gafanhoto{" + super.toString() + "login=" + login + ", totAssistido=" + totAssistido + '}';
    }
    
    //Métodos Acessores
    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public int getTotAssistido() {
        return totAssistido;
    }

    public void setTotAssistido(int totAssistido) {
        this.totAssistido = totAssistido;
    }
}

File: PHP/exercicios-PHPOO/Aula14e15/Gafanhoto.php
<?php
include_once 'Pessoa.php';
class Gafanhoto extends Pessoa
{
    //Atributos Privados
    private $login;
    private $totAssistido;

    //Método Especial
    public function viuMaisUm()
    {
        $this->totAssistido ++;
    }
    //Métodos Implementados da classe abstrata Pessoa
    protected function ganharXP($exp)
    {
        $this->experiencia += $exp;
    }
    //Método Construtor
    public function __construct($nome, $idade, $sexo, $login)
    {
        parent::__construct($nome, $idade, $sexo); //chamando o método construtor da classe mãe para herdar seus atributos
        $this->login = $login;
    }

    //Métodos Acessores
    public function getLogin()
    {
        return $this->login;
    }

    public function getTotAssistido()
    {
        return $this->totAssistido;
    }

    public function setLogin($login): void
    {
        $this->login = $login;
    }

    public function setTotAssistido($totAssistido): void
    {
        $this->totAssistido = $totAssistido;
    }
}
